Makefile

```diff
@@ -7,10 +7,38 @@ install:
 test:
 	dune runtest
 
-clean:
-	dune clean
-
 format:
 	dune build @fmt --auto-promote
 
-.PHONY: all gospel clean format test
+
+# To check that the markdown documentation is kept valid
+
+GOSPEL_CMD := dune exec --display=quiet -- bin/cli.exe
+DOCUMENTS := README.mli
+# DOCUMENTS += docs/docs/getting-started/first-spec.mli # duplicate definitions in OCaml blocks
+DOCUMENTS += docs/docs/language/constants-specifications.mli
+DOCUMENTS += docs/docs/language/function-contracts.mli
+DOCUMENTS += docs/docs/language/locations.mli
+DOCUMENTS += docs/docs/language/logical.mli
+DOCUMENTS += docs/docs/language/type-specifications.mli
+# DOCUMENTS += docs/docs/walkthroughs/fibonacci.mli # `let` in OCaml block
+DOCUMENTS += docs/docs/walkthroughs/mutable-queue.mli
+DOCUMENTS += docs/docs/walkthroughs/union-find.mli
+# DOCUMENTS += docs/src/pages/faq.mli # `...` in OCaml block
+CHECKS := $(patsubst %,%.check,$(DOCUMENTS))
+
+.PRECIOUS: $(DOCUMENTS)
+
+%.mli: %.md
+	awk '/^ *```ocaml/,/^ *```$$/ { if ($$0 !~ "^ *```") print }' < $< > $@
```

Comment:
I'm really bad at this so I won't try to fix it, but
```
val f : t
(*@ ... *)

val g : t
(*@ ... *)
```
seems to be rewritten as 
```
val f : t

(*@ ... *)
val g : t

(*@ ... *)
```
which could (should) be an incorrect attachment of the specs.